### ISSUE TITLE
haul fails when installed via pnpm

### ISSUE THREAD

**Issue by tachtevrenidis** [callstack/haul#432]
Has anybody tried haul via pnpm instead of yarn or npm? I am having a hard time getting it to work...

pnpm: https://pnpm.js.org/en/

STEPS TO REPRO

* mkdir test-dir
* pnpm install haul --save-dev
* node node_modules/haul/bin/cli.js

### Current Behavior

Error:
```
node ./node_modules/haul/bin/cli.js 
module.js:538
    throw err;
    ^

Error: Cannot find module 'babel-polyfill'
    at Function.Module._resolveFilename (module.js:536:15)
    at Function.Module._load (module.js:466:25)
    at Module.require (module.js:579:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/user/Development/pnpm-test/node_modules/.registry.npmjs.org/haul/1.0.0-rc.0/node_modules/haul/src/cli.js:10:1)
    at Module._compile (module.js:635:30)
    at Module._extensions..js (module.js:646:10)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/user/Development/pnpm-test/node_modules/.registry.npmjs.org/babel-register/6.26.0/node_modules/babel-register/lib/node.js:152:7)
    at Module.load (module.js:554:32)
    at tryModuleLoad (module.js:497:12)
```

### Expected Behavior
haul bundler starts running

### Haul Configuration (webpack.haul.js)
irrelevant for this issue

### Your Environment
| software         | version
| ---------------- | -------
| Haul             |  1.0.0-rc.0
| react-native     |  0.55.2
| node             |  8.9.3
| pnpmx      |  1.43.1

**Comment by ThomasMichon** [comment]
I've done some investigations and made local workarounds in my product's codebase, which can hopefully translate here:

1. I updated `package.json` to include `babel-polyfill` in `dependencies`. Since `cli.js` is doing
```js
require('babel-polyfill');
```
directly, I think it is reasonable to declare it as a direct dependency.

2. I had to modify the regular expressions used for `babel-register` and `makeReactNativeConfig` to account for the fact that `node_modules/` might occur *multiple times* in a file's path.

```js
/node_modules\/(?!react|@expo|pretty-format|haul|metro)/
```
becomes
```js
/node_modules(?!.*[\/\\](react|@expo|pretty-format|haul|metro))/
```
and
```js
/node_modules(?!\/haul)/
```
becomes
```js
/node_modules(?!.*[\/\\]haul)/
```
3. It turns out there is an issue with `react-native` as well, which requires `react` to be installed globally or by NPM directly in a higher-level folder. Specifically, `create-react-class`, which is used within `react-native`, does not properly declare a `peerDependency` on `react`.
Unfortunately Facebook appears to be rejecting requests to fix this: https://github.com/facebook/react/issues/11024

I believe the updates required to Haul are minor here: they all stem from assumptions about the idiosyncrasies of NPM.
